src/hotspot/share/gc/shared/collectedHeap.hpp

```diff
@@ -240,12 +242,15 @@ class CollectedHeap : public CHeapObj<mtGC> {
   virtual void post_initialize();
 
   // Stop any onging concurrent work and prepare for exit.
-  virtual void stop() {}
+  virtual void stop();
```

Comment:
I find it a bit odd to call `log_gc_vtime` inside this method, given the comment and its name. I wonder if `print_tracing_info` can be used instead, which is invoked before exit as well.

Reply:
I'd also prefer to see a small adjustment here. With this change all GCs have to add a back call to `CollectedHeap::stop()`. I think it makes the could a tiny bit less clean.

I wonder if an alternative could be to split `CollectedHeap::stop()` into two functions:
```
virtual void CollectedHeap::stop() = 0; // Now pure virtual

void CollectedHeap::before_exit() {
  // Log GC CPU usage.
  log_gc_vtime();

  // Stop any on-going concurrent work and prepare for exit.
  stop();
}
```
And revert the added calls to `CollectedHeap::stop()`.